app/models/stat.rb

```diff
@@ -0,0 +1,7 @@
+class Stat < ActiveRecord::Base
+	belongs_to :player
+	has_many :weight_stats
```

Comment:
We don't need the `has_many` relations here as well - weight_stats are already stats since it inherits from stats ;)